server/src/modules/campaign-monitor/campaign-monitor.service.ts

```diff
@@ -624,48 +629,72 @@ export default class CampaignMonitorService {
-		return writeFile(
+		return writeFileSync(
 			join(__dirname, `../../../data/${filename}.json`),
-			JSON.stringify(data),
-			(err) => {
-				if (err) {
-					throw new CustomError('Failed to create JSON file', {
-						filename,
-					});
-				}
-			}
+			JSON.stringify(data)
 		);
 	}
 
-	static async getAllLdapProfiles(): Promise<AvoLDAPProfile[]> {
+	static async getBatchLdapProfiles(offset = 0): Promise<AvoLdapPaginatedProfile> {
 		try {
-			const response = await DataService.execute(GET_ALL_LDAP_PROFILES);
-			const profiles = response.data.users_idp_map.map(
-				(users_idp: { local_user: { profiles: AvoLDAPProfile[]; id: number } }) => {
-					const sharedUser = response.data.shared_users.find(
-						(u: {
-							id: number;
-							mail: string;
-							idpmaps: { idpByIdp: { value: string } }[];
-						}) => u.id === users_idp.local_user?.id
-					);
-					return {
-						...users_idp.local_user?.profiles?.[0],
-						email: sharedUser.mail,
-						links: sharedUser?.idpmaps?.map(
-							(idp: { idpByIdp: { value: string } }) => idp.idpByIdp.value
-						),
-					};
+			const response: { data: AvoLDAPProfileData } = await DataService.execute(
+				GET_ALL_LDAP_PROFILES,
+				{
+					offset,
+					limit: avoListSyncBatchSize,
 				}
 			);
-			this.createJSONFile(profiles, `avo`);
-
-			return profiles;
+			return {
+				items: response.data.users_idp_map.map((users_idp) => ({
+					...users_idp?.local_user?.profiles?.[0],
+					id: users_idp?.local_user?.id,
+					is_blocked: users_idp?.local_user?.is_blocked,
+					last_access_at: users_idp?.local_user?.last_access_at,
+					temp_access: users_idp?.local_user?.temp_access,
+					last_block: users_idp?.local_user?.last_block,
+					idpmaps: users_idp?.local_user?.idpmaps,
+					mail: users_idp?.local_user?.mail,
+				})),
+				total: response.data.users_idp_map_aggregate.aggregate.count,
+			};
 		} catch (err) {
 			throw new CustomError('Failed to fetch ldap users from avo database', err);
 		}
 	}
 
+	static async getAllLdapProfiles(): Promise<void> {
+		return this.getBatchLdapProfiles()
+			.then((initialResponse: AvoLdapPaginatedProfile) => {
+				if (initialResponse.total <= avoListSyncBatchSize) {
+					return Promise.resolve(initialResponse.items);
+				}
+
+				// Based on the total items, we do x amount of requests
+				const promises = times(
+					(i: number) =>
+						i === 0
+							? Promise.resolve(initialResponse)
+							: this.getBatchLdapProfiles(i * 1),
+					Math.ceil(initialResponse.total / avoListSyncBatchSize)
+				);
+
+				// We take the results and combine them into one array without pagination
+				return Promise.all(promises).then((responses: AvoLdapPaginatedProfile[]) =>
+					responses.reduce(
+						(
+							acc: (Omit<AvoLDAPProfile, 'id'> & AvoLDAPProfileDataLocalUserProps)[],
+							response: AvoLdapPaginatedProfile
+						) => {
+							return [...acc, ...response.items];
+						},
+						[]
+					)
+				);
```

Comment:
Isn't this the same thing?
I'm not a big fan of reduce. :)
```suggestion
				const responses: AvoLdapPaginatedProfile[] = await Promise.all(promises);
				return responses.map(r => r.items).flat();
```